apidef/oas/server.go

```diff
@@ -33,6 +36,11 @@ type Server struct {
 	//
 	// Tyk classic API definition: `detailed_tracing`
 	DetailedTracing *DetailedTracing `bson:"detailedTracing,omitempty" json:"detailedTracing,omitempty"`
+
+	// Events contains the configuration related to Tyk Events.
+	//
+	// Tyk classic API definition: `event_handlers`
+	Events Events `bson:"events,omitempty" json:"events,omitempty"`
```

Comment:
It's a slice, so a pointer wouldn't be necessary. like `CustomPlugins`

Reply:
I agree it's not necessary, it's only a readability/complexity hint:

- the field is optional in json
- pointer is an explicit optionality hint (human readable, no lookups)

The problems are: is Events a concrete type or interface confusion (it's uncommon to have interfaces prefixed with *ptr, resolving the confusion). Without the explicit pointer reference, it's harder because we have to look up the type, and also explicit handling of defaults is more complex, and mistakes are made asuming the fields will have a deep copy due to the implicit pointer, and the explicit pointer gives you a readable hint that this isn't the case.

Feel free to resolve the suggestion and decide if to apply or not. It only carries some semantic value here